src/utils/IdealGraphVisualizer/View/src/main/java/com/sun/hotspot/igv/view/DiagramScene.java

```diff
@@ -471,38 +297,171 @@ public void select(Widget widget, Point localLocation, boolean invertSelection)
+                                Sheet s = super.createSheet();
+                                PropertiesSheet.initializeSheet(provider.getProperties(), s);
+                                return s;
+                            }
+                        };
+                        node.setDisplayName(provider.getProperties().get("name"));
+                        content.add(node);
+                    }
+
+
+                    if (o instanceof Figure) {
+                        nodeSelection.add(((Figure) o).getInputNode().getId());
+                    } else if (o instanceof Slot) {
+                        nodeSelection.addAll(((Slot) o).getSource().getSourceNodesAsSet());
+                    }
+                }
+                getModel().setSelectedNodes(nodeSelection);
+
+                boolean b = selectedCoordinatorListener.isEnabled();
+                selectedCoordinatorListener.setEnabled(false);
+                SelectionCoordinator.getInstance().setSelectedObjects(nodeSelection);
+                selectedCoordinatorListener.setEnabled(b);
+
+            }
+
+            @Override
+            public void highlightingChanged(ObjectSceneEvent e, Set<Object> oldSet, Set<Object> newSet) {
+                Set<Integer> nodeHighlighting = new HashSet<>();
+                for (Object o : newSet) {
+                    if (o instanceof Figure) {
+                        nodeHighlighting.add(((Figure) o).getInputNode().getId());
+                    } else if (o instanceof Slot) {
+                        nodeHighlighting.addAll(((Slot) o).getSource().getSourceNodesAsSet());
+                    }
+                }
+                highlightedCoordinatorListener.setEnabled(false);
+                SelectionCoordinator.getInstance().setHighlightedObjects(nodeHighlighting);
+                highlightedCoordinatorListener.setEnabled(true);
+            }
+
+            @Override
+            public void hoverChanged(ObjectSceneEvent e, Object oldObject, Object newObject) {
+                Set<Object> newHighlightedObjects = new HashSet<>(getHighlightedObjects());
+                if (oldObject != null) {
+                    newHighlightedObjects.remove(oldObject);
+                }
+                if (newObject != null) {
+                    newHighlightedObjects.add(newObject);
+                }
+                setHighlightedObjects(newHighlightedObjects);
+            }
+
+            @Override
+            public void focusChanged(ObjectSceneEvent arg0, Object arg1, Object arg2) {
+            }
+        };
+        addObjectSceneListener(selectionChangedListener, ObjectSceneEventType.OBJECT_SELECTION_CHANGED, ObjectSceneEventType.OBJECT_HIGHLIGHTING_CHANGED, ObjectSceneEventType.OBJECT_HOVER_CHANGED);
     }
 
     public DiagramViewModel getModel() {
         return model;
     }
 
-    public JScrollPane getScrollPane() {
+    private JScrollPane getScrollPane() {
         return scrollPane;
     }
```

Comment:
You could directly replace the usages of `getScrollPane()` inside this class by a field access to `scrollPane`.

Reply:
done